### ISSUE TITLE
Incorrectly passing code for the `Style/BlockDelimiters` rule

### ISSUE THREAD

**Issue by revolter** [rubocop-hq/rubocop#6179]
Using a `lambda` block doesn't trigger an error for the `Style/BlockDelimiters` rule when used incorrectly (curly braces for multiline code block). The `fails.rb` file correctly fails with an error for the `[].each { ... }` code block.

--------

## Expected behavior

To fail when running against the `passes.rb` file.

## Actual behavior

It passes with no errors.

## Steps to reproduce the problem

`passes.rb`:

```
lambda {
  puts('test')
}

lambda do
  puts('test')
end
```

`fails.rb`:

```
[].each {
  puts('test')
}

[].each do
  puts('test')
end

```

## RuboCop version

Include the output of `rubocop -V`. Here's an example:

```
$ rubocop -V
0.58.2 (using Parser 2.5.1.2, running on ruby 2.3.7 universal.x86_64-darwin17)
```

**Comment by revolter** [comment]
Accidentally hit send. Will edit the issue and delete this comment.

**Comment by koic** [comment]
Thanks for the feedback.

Since `lambda` is included in `IgnoredMethods` default, it seems to appear as a false negative.
https://github.com/rubocop-hq/rubocop/blob/v0.58.2/config/default.yml#L902-L918

`IgnoredMethods` is provided for `EnforcedStyle: semantic`.
https://github.com/rubocop-hq/rubocop/commit/36faf0165626190e50b09a92753298ff50ac9aaf

But `IgnoredMethods` has been applied to all `EnforcedStyle`. Therefore, it also applies to the default `EnforcedStyle: line_count_based`.
It seems that the difference shown in the reproduction feedback is caused by this fact.

IMO. The following is an idea to solve this problem.

`EnforcedStyle: line_count_based` can determine how to use `{...}` and `do..end` by the difference between single-line and multi-line. So when it is `EnforcedStyle: line_count_based` it may be better to ignore the` IgnoredMethod` setting.
I think that it will not be a big problem, but this idea that will be breaking change.
